src/SilverpopConnector/Xml/XmlAction.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace SilverpopConnector\Xml;
+
+/**
+ * Created by IntelliJ IDEA.
+ * User: emcnaughton
+ * Date: 5/10/17
+ * Time: 8:23 AM
+ */
+class XmlAction {
+
+  public function __construct($params) {
```

Comment:
Shouldn't this class be declared as `abstract`?

In fact, given that all of the child classes need a `formatResult()` method, adding that as an abstract method here would help to make the expected interface for the various child classes more clear.